eclair-core/src/test/scala/fr/acinq/eclair/channel/states/h/ClosingStateSpec.scala

```diff
@@ -1370,13 +1370,18 @@ class ClosingStateSpec extends TestKitBaseClass with FixtureAnyFunSuiteLike with
     // bob manages to claim 2 htlc outputs before alice can penalize him: 1 htlc-success and 1 htlc-timeout.
     val (fulfilledHtlc, preimage) = revokedCloseFixture.htlcsAlice.head
     val (failedHtlc, _) = revokedCloseFixture.htlcsBob.last
-    val bobHtlcSuccessTx1 = bobRevokedTxs.htlcTxsAndSigs.collectFirst {
-      case HtlcTxAndSigs(txInfo: HtlcSuccessTx, localSig, remoteSig) if txInfo.htlcId == fulfilledHtlc.id =>
+    val bobKeyManager = bob.underlyingActor.nodeParams.channelKeyManager
+    val bobChannelKeyPath = bobKeyManager.keyPath(bob.stateData.asInstanceOf[DATA_NORMAL].commitments.localParams, channelVersion)
+    val bobPerCommitmentPoint = bobKeyManager.commitmentPoint(bobChannelKeyPath, bobRevokedCommit.index.toInt)
+    val bobHtlcSuccessTx1 = bobRevokedCommit.htlcTxsAndRemoteSigs.collectFirst {
+      case HtlcTxAndRemoteSig(txInfo: HtlcSuccessTx, remoteSig) if txInfo.htlcId == fulfilledHtlc.id =>
         assert(fulfilledHtlc.paymentHash === txInfo.paymentHash)
+        val localSig = bobKeyManager.sign(txInfo, bobKeyManager.htlcPoint(bobChannelKeyPath), bobPerCommitmentPoint, TxOwner.Local, commitmentFormat)
         Transactions.addSigs(txInfo, localSig, remoteSig, preimage, commitmentFormat)
     }.get
-    val bobHtlcTimeoutTx = bobRevokedTxs.htlcTxsAndSigs.collectFirst {
-      case HtlcTxAndSigs(txInfo: HtlcTimeoutTx, localSig, remoteSig) if txInfo.htlcId == failedHtlc.id =>
+    val bobHtlcTimeoutTx = bobRevokedCommit.htlcTxsAndRemoteSigs.collectFirst {
+      case HtlcTxAndRemoteSig(txInfo: HtlcTimeoutTx, remoteSig) if txInfo.htlcId == failedHtlc.id =>
+        val localSig = bobKeyManager.sign(txInfo, bobKeyManager.htlcPoint(bobChannelKeyPath), bobPerCommitmentPoint, TxOwner.Local, commitmentFormat)
```

Comment:
You don't need all this logic, these transactions don't need to be signed with valid signatures, they only need to contain a script witness with the right format (and the preimage).

You can remove all this and change the code to simply do:
```scala
// HTLC-success
Transactions.addSigs(txInfo, ByteVector64.Zeroes, ByteVector64.Zeroes, preimage, commitmentFormat)
// HTLC-timeout
Transactions.addSigs(txInfo, ByteVector64.Zeroes, ByteVector64.Zeroes, commitmentFormat)
```

Which will help us focus on the important parts of the test.
It's my bad, I wrote that code and could have made it clear that we don't need the actual signatures, just something that looks like a signature.

Reply:
Done in 2e074f710 (squashed).